packages/contracts/contracts/optimistic-ethereum/chain/CanonicalTransactionChain.sol

```diff
@@ -21,6 +20,12 @@ contract CanonicalTransactionChain {
     bytes32[] public batches;
     uint public lastOVMTimestamp;
 
+    /*
+     * Events
+     */
+
+    event QueueBatchAppended( bytes32 _batchHeaderHash, bytes32 _txHash);
```

Comment:
I think that we shouldn't need to emit the tx hash for a couple reasons.

1) the `txHash` is the hash of the *rollup transaction*, not the hash of the L1 transaction. That means it's not useful for something like `getTransactionByHash` calls using the web3 API.
2) if we assume that we're running an L1 full-node, we should have parsed all blocks and be able to get a transaction by it's `batchHeaderHash`

So yeah I actually don't know what the txHash would be used for because it's not really helping us retrieve the tx data